tools/busco/main.nf

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env nextflow
+
+// Specify DSL2
+nextflow.preview.dsl = 2
+
+// Process definition
+process busco_genome {
+    publishDir "${params.outdir}/${opts.publish_dir}",
+        mode: "copy",
+        overwrite: true,
+        saveAs: { filename ->
+                      if (opts.publish_results == "none") null
+                      else filename }
+
+    container "ezlabgva/busco:v4.1.3_cv1"
+
+    input:
+        val opts
+        tuple val(meta), path(genome)
+
+    output:
+        tuple val(meta), path("busco"), emit: busco_genome_out
```

Comment:
we had a variable name meeting (I know), can you change the name to describe the files emitted? our preferred names are things like `sam` and `bam` etc.

Reply:
I would like some clarification for this comment. Is the goal to make the `emit` names descriptive (describing the files, as you say), or is the goal to make them compact and simple? 

A simple `emit` name is fine if the process name is specific since the distinction between related outputs will be clear (e.g., `initial_alignment.out.bam` vs. `highly_modified_and_subsetted_alignment.out.bam`). But if both the process name and the emit name are generic and short, it will be harder to distinguish outputs that happen to share an output type (e.g., `align.out.bam` vs. `align.out.bam`).

In this case the output of the process is a directory containing a bunch of files and subdirectories. I could split every expected subdirectory and file into separate `emit` statements, but there are many, and none of outputs will be specifically used as input to later processes. The purpose of this module is essentially QC; while none of its outputs are particularly useful, it is necessary to generate them and they are worth keeping. 

Having described this, is it better for the emit name to be simply _busco_, or something like _busco_genome_output_directory_?

Reply:
well based on what you are saying, cant you just call it `qc`? Or, if its a reporting output that is not consumed by a downstream process, we usually call this `report`